clang/test/Analysis/stack-addr-ps.cpp

```diff
@@ -791,3 +791,30 @@ void global_ptr_to_ptr() {
   *global_pp = nullptr;
 }
 } // namespace leaking_via_indirect_global_invalidated
+
+namespace not_leaking_via_simple_ptr {
+void top(const char *p) {
```

Comment:
Please add a ref case too.
And also one for taking a struct by value and having this pointer as a member, and a similar with ref.

Reply:
I am not sure what you are referring to by the "ref case". There is already `param_ref_to_struct_with_ptr_top` and `param_ref_to_ptr_top`.
I added other cases in 8a9a76a3d1b1